### ISSUE TITLE
cDOT 8.3.2

### ISSUE THREAD

**Issue by ikkaro** [puppetlabs/puppetlabs-netapp#68]
It seems that hostname is not working in version 8.3.2, I think that the hostname is not retreived correctly.

Debug: System info = {"version"=>"NetApp Release 8.3.2: Wed Feb 24 03:29:11 UTC 2016", "clustered"=>"true", "hostname"=>nil}

**Comment by amit0701** [comment]
To support HA, the hostname field is populated with the "cluster name" of the cDOT system and not the individual hostnames of the nodes. Having said that, the hostname  field should not be nil.

Are you using the latest version of the module 1.1.0 ?  

I would verify the cluster name specified in the device.conf is resolved to the cDOT cluster.

Some documentation about the module:
http://www.netapp.com/us/media/tr-4477.pdf

**Comment by ikkaro** [comment]
Hi,

I'm using the cluster hostname, and not the controllers hostnames.
This is my device.conf

[tv1-na03.mydomain.com]
type
netapp                                 
url https://USER:PASS@tv1-na03.mydomain.com

This is the output from a ssh console, excuting the hostname command. As you note there is a blank line in the output. 
tv1-na03::> hostname 
tv1-na03

tv1-na03::> 

I'm using version 1.1.0, and also the hostname is resolving the an ip address.

**Comment by amit0701** [comment]
Hello,

Did you get an error "No matching system found with the system name tv1-na03.mydomain.com" ?

Also I see that line 141  in 
https://github.com/puppetlabs/puppetlabsnetapp/blob/master/lib/puppet/util/network_device/netapp/facts.rb should be patched. So that it checks if the hostname specified matches against the cluster name retrieved.(The if block only checks for node names)

I am linking my personal  facts.rb . Let me know if it works. If so, I will make a formal pull request and push it upstream

My attempt at patching facts.rb : 
https://github.com/amit0701/DataONTAPDevice/blob/master/lib/puppet/util/network_device/netapp/facts.rb

**Comment by lm-lamb** [comment]
@amit0701 I'm getting the error you mentioned ("No matching system found with the system name... "and haven't been able to figure anything out. My device.conf file is setup correctly. Unfortunately, it doesn't seem that your patch will fix the problem. 

system_host is set to the cluster name if it matches the host name on line 100. So 141 is already set to the cluster name if it should be (this would need to happen before the if statement on line 122, as it does). Your patch shouldn't be necessary because this assignment was already done, and it removes the error handling that was supposed to occur if system_host was not set.

Any other suggestions?

**Comment by amit0701** [comment]
A few pull requests were recently merged, did u try with the updated code from the repo?  let me know if you still encounter the issue with the latest code. 

if it still doesn't solve your issue, please shoot me an email(amit.borulkar@netapp.com) we can schedule something and look at the device.conf and the debug output
